src/Reflection/ReflectionParameter.php

```diff
@@ -366,4 +375,23 @@ public function getDefaultValueConstantName()
 
         return $this->defaultValueConstantName;
     }
+
+    /**
+     * Gets a ReflectionClass for the type hint (returns null if not a class)
+     *
+     * @return ReflectionClass|null
+     */
+    public function getClass()
+    {
+        $hint = $this->getTypeHint();
+        if (!($hint instanceof Types\Object_)) {
+            return null;
+        }
+
+        if (!$this->reflector instanceof ClassReflector) {
```

Comment:
This becomes an issue now because if this is a parameter that belongs to a function, the `$this->reflector` will be a `FunctionReflector` and thus we'll never be able to reflect on classes that are typehinted in a `ReflectionFunction`... :/